crates/atiny-checker/src/context.rs

```diff
@@ -47,13 +48,11 @@ impl Default for Ctx {
 
         ctx.signatures.types.insert(
             "Int".to_string(),
-            TypeSignature {
-                name: "Int".to_string(),
-                params: vec![],
-                value: TypeValue::Opaque,
-            },
+            TypeSignature::new_opaque("Int".to_string()),
         );
 
+        ctx.extend_type_sigs(iter::once(TypeDecl::unit()));
```

Comment:
Yeah it's just a matter of design but remember we can just keep unit as a type defined type with `()` being a syntax sugar for `Unit`